src/Microsoft.OData.Edm/Validation/ExpressionTypeChecker.cs

```diff
@@ -486,6 +488,31 @@ private static bool TryCastBinaryConstantAsType(IEdmBinaryConstantExpression exp
             return true;
         }
 
+        private static bool TryCastEnumConstantAsType(IEdmEnumMemberExpression expression, IEdmTypeReference type, bool matachExactly, out IEnumerable<EdmError> discoveredErrors)
+        {
+            if (!type.IsEnum())
+            {
+                discoveredErrors = new EdmError[]
+                {
+                    new EdmError(expression.Location(),
+                    EdmErrorCode.ExpressionEnumKindNotValidForAssertedType,
+                    Edm.Strings.EdmModel_Validator_Semantic_ExpressionEnumKindNotValidForAssertedType)
+                };
+                return false;
+            }
+
+            foreach (var member in expression.EnumMembers)
+            {
+                if (!TestTypeMatch(member.DeclaringType, type.Definition, expression.Location(), matachExactly, out discoveredErrors))

```

Comment:
I see that the typo "matachExactly" came from somewhere else. I think it should be fixed there too.

Reply:
#resolved.